processor/k8sprocessor/options.go

```diff
@@ -249,3 +249,27 @@ func WithExtractPodAssociations(podAssociations ...PodAssociationConfig) Option
 		return nil
 	}
 }
+
+// WithIgnoredPodNames allows specifying pod names to ignore
+func WithIgnoredPodNames(podIgnore PodIgnoredConfig) Option {
+	return func(p *kubernetesprocessor) error {
+		ignoredNames := kube.IgnoredPodNames{}
+		defaultIgnoredNames := []*regexp.Regexp{
+			regexp.MustCompile(`jaeger-agent`),
+			regexp.MustCompile(`jaeger-collector`),
+		}
+
+		if len(podIgnore.Names) == 0 {
+			ignoredNames.Regex = defaultIgnoredNames
+			p.podIgnore = ignoredNames
+			return nil
+		}
+
+		for _, name := range podIgnore.Names {
+			r := regexp.MustCompile(name)
+			ignoredNames.Regex = append(ignoredNames.Regex, r)
+		}
+		p.podIgnore = ignoredNames
```

Comment:
This allocates a `[]*regexp.Regexp` and the two default regexes, even if they're not going to be used.  With a slight rearrangement that allocation and initialization can be avoided:

```suggestion
		ignoredNames := kube.IgnoredPodNames{}
		names := podIgnore.Names

		if len(names) == 0 {
			names = []*regexp.Regexp{
				regexp.MustCompile(`jaeger-agent`),
				regexp.MustCompile(`jaeger-collector`),
			}
		}

		for _, name := range names {
			ignoredNames.Regex = append(ignoredNames.Regex, regexp.MustCompile(name))
		}
		p.podIgnore = ignoredNames
```

Reply:
Thanks! Other than names being of type []string, the new commit should reflect your comment. :)